src/pcm/SoxrResampler.cxx

```diff
@@ -65,6 +85,104 @@ soxr_quality_name(unsigned long recipe) noexcept
 	}
 }
 
+static char *
+quality_advanced_nextparam(char *src, char c)
+{
+        static char *str = NULL;
+        char *ptr, *ret;
+
+        if (src)
+		str = src;
+        if (str && (ptr = strchr(str, c))) {
+                ret = str;
+                *ptr = '\0';
+                str = ptr + 1;
+        } else {
+                ret = str;
+                str = NULL;
+        }
+
+        return ret && ret[0] ? ret : NULL;
+}
+
+static bool
+soxr_parse_quality_advanced(const char *copt, struct soxr *r)  noexcept
+{
+	char *opt, *recipe = NULL, *flags = NULL;
+	char *atten = NULL;
+	char *precision = NULL, *passband_end = NULL, *stopband_begin = NULL, *phase_response = NULL;
+
+	opt = xstrdup(copt);
```

Comment:
Do not use `xstrdup()`, it's an error prone C function, leading to memory leaks.